orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -0,0 +1,217 @@
+
+    def forward(self, model_input, x=None, y=None, z=None):
+        model_input = model_input + torch.sum(torch.stack([x, y, z]), dim=0)
+        out = self.fc1(model_input)
+        out = self.relu(out)
+        out = self.fc2(out)
+        return out
+
+# ORTModule-API tests
+
+def test_forward_call_single_argument():
+    device = 'cuda'
+
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetSingleArgument(D_in, H, D_out).to(device)
+    model = ORTModule(model)
+    x = torch.randn(N, D_in, device=device)
+    try:
+        model(x)
+    except Exception as exception:
+        raise exception
+
+def test_forward_call_multiple_arguments():
+    device = 'cuda'
+
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetMultipleArguments(input_size=D_in, hidden_size=H, num_classes=D_out).to(device)
+    model = ORTModule(model)
+    x = torch.randn(N, D_in, device=device)
+    y = torch.randn(N, D_in, device=device)
+    try:
+        model(x, y)
+    except Exception as exception:
+        raise exception
+
+def test_forward_call_positional_arguments():
+    device = 'cuda'
+
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetPositionalArguments(input_size=D_in, hidden_size=H, num_classes=D_out).to(device)
+    model = ORTModule(model)
+    args = [torch.randn(N, D_in, device=device), torch.randn(N, D_in, device=device), torch.randn(N, D_in, device=device)]
+    try:
+        model(*args)
+    except Exception as exception:
+        raise exception
+
+def test_forward_call_keyword_arguments():
+    device = 'cuda'
+
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetKeywordArguments(D_in, H, D_out).to(device)
+    model = ORTModule(model)
+    x = torch.randn(N, D_in, device=device)
+    y = torch.randn(N, D_in, device=device)
+    z = torch.randn(N, D_in, device=device)
+    try:
+        model(x, y, z)
+    except Exception as exception:
+        raise exception
+
+def test_forward_call_positional_and_keyword_arguments():
+    device = 'cuda'
+
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetPositionalAndKeywordArguments(D_in, H, D_out).to(device)
+    model = ORTModule(model)
+    a = torch.randn(N, D_in, device=device)
+    x = torch.randn(N, D_in, device=device)
+    y = torch.randn(N, D_in, device=device)
+    z = torch.randn(N, D_in, device=device)
+    try:
+        model(a, x, y, z)
+    except Exception as exception:
+        raise exception
+
+def test_model_cuda():
+    device = 'cuda'
+
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = NeuralNetSingleArgument(D_in, H, D_out)
+    model = ORTModule(model)
+    x = torch.randn(N, D_in, device=device)
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cpu'
+
+    model = model.cuda()
+    model(x)
+
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cuda'
+
+def test_model_cpu():
+    device = 'cuda'
```

Comment:
Ha!

Reply:
The ```device``` here is just for using a string literal. Basically, the model starts off as a cuda one. And then I call ```cpu``` on it and check whether the named_parameters have been migrated to cpu. I can leave a comment for this.